src/react_native/biometrics.cljs

```diff
@@ -0,0 +1,76 @@
+(ns react-native.biometrics
+  (:require
+    ["react-native-biometrics" :default biometrics]
+    [clojure.string :as string]
+    [oops.core :as oops]
+    [react-native.platform :as platform]
+    [schema.core :as schema]))
+
+(def ^:const biometric-error-not-available "biometric-not-available")
+(def ^:const biometric-error-not-enrolled "biometric-not-enrolled")
+
+(defn get-supported-type
+  "Returns a JS promise that resolves with the biometrics types supported by the
+  device, regardless of whether it's enabled or disabled.
+
+  Resolved values: `:Biometrics` `:FaceID` `:TouchID`"
+  []
+  (-> (biometrics.)
+      (.isSensorAvailable)
+      (.then (fn [result]
+               (let [type (-> result
+                              (oops/oget "biometryType")
+                              keyword)]
+                 type)))
+      (.catch (constantly nil))))
+
+(defn get-available
+  "Returns a JS promise that resolves to a boolean, which signifies whether
+  biometrics is enabled/disabled on the device."
+  []
+  (-> (biometrics.)
+      (.isSensorAvailable)
+      (.then (fn [result]
+               (oops/oget result "available")))))
+
+;; NOTE: the react-native-biometry package error codes/messages differ across platforms.
+;; On android we get error messages while on iOS it's a stringified Obj-C error object.
+(defn- convert-auth-error-message
+  [code]
+  (letfn [(includes? [substring string]
+            (string/includes? string substring))]
+    (if platform/android?
+      (condp = code
+        "No fingerprints enrolled."      biometric-error-not-enrolled
+        "Biometric hardware unavailable" biometric-error-not-available
+        code)
+
+      (condp includes? code
+        "No identities are enrolled" biometric-error-not-enrolled
+        code))))
```

Comment:
The error messages coming from `react-native-biometrics` are inconsistent across platforms. On iOS it just rejects with a stringified Obj-C error object and on android it's sentences with or without a period.